core-bundle/src/Config/Loader/PhpFileIncluder.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Config\Loader;
+
+use Symfony\Component\Config\Loader\Loader;
+
+/**
+ * Includes the given php file, e.g. for old php/GLOBALS based translations
+ *
+ * @author Mike vom Scheidt <https://github.com/mvscheidt>
+ */
+class PhpFileIncluder extends Loader
+{
+    /**
+     * Includes a php file
+     *
+     * @param string      $file
+     * @param string|null $type
+     *
+     * @return void
+     */
+    public function load($file, $type = null)
+    {
+        $this->includePhp($file);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports($resource, $type = null)
+    {
+        return 'php' === pathinfo((string) $resource, PATHINFO_EXTENSION);
+    }
+
+    /**
+     * includes the given file
+     *
+     * @param $file
+     */
+    private function includePhp($file)
+    {
+        include $file;
```

Comment:
Should I leave this like this or instead do something like

```
if ((@include $file) === false)
{
    throw new Exception(...)
}
```

The issue I see with this is that I found no way to discern between different problems that can occur when trying to include a file, we only now if it failed - we do not know if it means that

- file did not exist
- file was not valid php
- file not readable

I guess I could do this inside the support method, however then the `loadLanguageFile` needs some more work because otherwise we would do the "file exists" check twice.

Reply:
> The issue I see with this is that I found no way to discern between different problems that can occur when trying to include a file, we only know if it failed

Exactly. Therefore I'm in favor of keeping the `include $file` statement, too.